### PR TITLE
Use helper function to compare guid

### PR DESCRIPTION
**Pull request by crobibero**
I found a few places where we were comparing a nullable guid to `default`, which isn't what was actually desired.

Double check to make sure I didn't replace anything in an EF call.

### CODE REVIEW

**Comment by barronpm** [comment]
If this is Guid specific, does it make sense to name them `IsEmpty` and `IsNullOrEmpty`? IMO that's a bit clearer in what the intent is behind the comparisons.

File: MediaBrowser.Controller/Entities/Video.cs [hunk 1]
```diff
@@ -456,7 +456,7 @@ private void RefreshLinkedAlternateVersions()
             foreach (var child in LinkedAlternateVersions)
             {
                 // Reset the cached value
-                if (child.ItemId.HasValue && child.ItemId.Value.Equals(default))
+                if (child.ItemId.IsNullOrEmpty())
```
Comment (barronpm) [inline MediaBrowser.Controller/Entities/Video.cs:459]: This changes behavior, the existing code checks that it is empty but _not_ `null`

Comment (crobibero) [reply MediaBrowser.Controller/Entities/Video.cs:459]: I don't think the difference is meaningful enough to keep the `.HasValue` check, now a Guid? that is already null can be set to null again

Comment (barronpm) [reply MediaBrowser.Controller/Entities/Video.cs:459]: Didn't see the body of the if, makes sense

**Review by barronpm — APPROVED** [review]

**Pull request merged**